modules/core/src/commonMain/kotlin/fr/acinq/lightning/NodeParams.kt

```diff
@@ -272,4 +275,26 @@ data class NodeParams(
         return OfferManager.deterministicOffer(chainHash, nodePrivateKey, trampolineNodeId, amount, description, nonce)
     }
 
+    /**
+     * Generate a compact Bolt 12 offer based on the node's seed and its trampoline node.
+     * A compact offer is smaller since it allows the encryptedData payload to be empty for
+     * the final recipient (us) within the blinded path. This allows it to fit in restricted
+     * spaces (e.g. bolt card)
+     *
+     * Each generated offer is unique, since a random nonce is used to generate the blindedPathSessionKey.
+     * This offer will stay valid after restoring the seed on a different device: we will
+     * automatically keep accepting payments for this offer.
```

Comment:
This doesn't seem compatible with the requirement to store the key in `compactOfferKeys` which will not be restored from the seed.